boa_engine/src/bytecompiler/mod.rs

```diff
@@ -1180,7 +1171,6 @@ impl<'ctx, 'host> ByteCompiler<'ctx, 'host> {
     }
 
     /// Compile a [`Declaration`].
-    #[allow(unused_variables)]
```

Comment:
This is showing warnings because `block` is not used if the feature `annex-b` is disabled. Maybe rename `block` to `_block`?

Reply:
Renaming to `_block` does not seem to work as we get a different lint for usage of `_*` named variables. I just added the `allow` back.